var/spack/repos/builtin/packages/fckit/package.py

```diff
@@ -39,6 +39,7 @@ class Fckit(CMakePackage):
     depends_on("llvm-openmp", when="+openmp %apple-clang", type=("build", "run"))
     variant('shared', default=True)
     variant("fismahigh", default=False, description="Apply patching for FISMA-high compliance")
+    variant('badfinal', default=False, description='Forcibly disable automatic finalization of derived types')
```

Comment:
I wonder if instead of a double negative (... disable ... default off) use something like enable default true and name the variant something like `finalize_ddts` or so?

Reply:
How about this?
```python
    variant("finalize_ddts", default="auto",
            description="Enable / disable automatic finalization of derived types",
            values=("auto", "no", "yes"))

    def cmake_args(self):
        ......
        if 'finalize_ddts=auto' not in self.spec:
            args.append(self.define_from_variant('ENABLE_FINAL', 'finalize_ddts'))
```

Reply:
Yes, that's a lot better. Last question is if you wanted to have add a conflict statement for the problematic case that you identified.

Reply:
We are unfortunately not running the tests as part of the installs, hmm. Anyway, it's ok, we can add this for now. 

Thanks for the PR!